light-client/src/predicates.rs

```diff
@@ -120,6 +120,24 @@ pub trait VerificationPredicates: Send {
         Ok(())
     }
 
+    /// Check that the untrusted header is from past.
+    fn is_header_from_past(
+        &self,
+        header: &Header,
+        clock_drift: Duration,
+        now: Time,
+    ) -> Result<(), VerificationError> {
+        ensure!(
+            header.time < now + clock_drift,
+            VerificationError::HeaderFromTheFuture {
+                header_time: header.time,
+                now
+            }
+        );
```

Comment:
Isn't that already ensured in `is_within_trust_period`?

Reply:
I think we should remove it from `is_within_trust_period` because the header is already trusted. What do you think?

Reply:
I think it make sense to remove this check. With the new check you introduced (`is_header_from_past`), it should be guaranteed that we do not trust and store headers from the future. Hence, the check on a already trusted header seems redundant.